app/geo/gui/src/ts/twxplore/gui/geo/reducerFunctions/updateAttributeStatesOfFeatureType.ts

```diff
@@ -11,34 +14,66 @@ export const updateAttributeStatesOfFeatureType = (
   //Loop through the feature's attribute
   for (const attributeName of Object.keys(addedFeature)) {
     const attributeKey = attributeName as keyof MapFeature;
+    //Specifying the exact attribute state to be updated.
     let attributeStateOfFeatureType =
       attributeStatesOfFeatureType[attributeName];
-    //If the arttribute is numeric
-    if (getFeatureAttributeByName(attributeName).isNumeric) {
-      //If this is the first time coming across this attribute for this feature type
-      if (
-        attributeStateOfFeatureType.min === null &&
-        attributeStateOfFeatureType.max === null
-      ) {
-        //New max of the attribute wil be the feature's value for the attribute
-        attributeStateOfFeatureType.max = addedFeature[attributeKey] as number;
-        //New min of the attribute wil be the feature's value for the attribute
-        attributeStateOfFeatureType.min = addedFeature[attributeKey] as number;
-        continue;
+    switch (getFeatureAttributeByName(attributeName).typeOf) {
+      //If the attribute is numeric
+      case TypeOfFeatureAttribute.NUMBER: {
```

Comment:
Another good place to delegate to the strategy. featureAttribute...buildInitialState(...)